crates/ra_editor/src/lib.rs

```diff
@@ -97,13 +106,37 @@ pub fn diagnostics(file: &SourceFileNode) -> Vec<Diagnostic> {
         }
     }
 
-    file.errors()
+    let mut errors: Vec<Diagnostic> = file
+        .errors()
         .into_iter()
         .map(|err| Diagnostic {
             range: location_to_range(err.location()),
             msg: format!("Syntax Error: {}", err),
+            severity: Some(Severity::Error),
         })
-        .collect()
+        .collect();
+
+    let warnings = check_unnecessary_braces_in_use_statement(file);
+
+    errors.extend(warnings);
+    errors
+}
+
+fn check_unnecessary_braces_in_use_statement(file: &SourceFileNode) -> Vec<Diagnostic> {
+    let mut diagnostics = Vec::new();
+    for node in file.syntax().descendants() {
+        if let Some(use_tree_list) = ast::UseTreeList::cast(node) {
+            if use_tree_list.use_trees().count() <= 1 {
+                diagnostics.push(Diagnostic {
+                    range: use_tree_list.syntax().range(),
+                    msg: format!("Unnecessary braces in use statement"),
```

Comment:
Ideally, this would have a suggestion of how to fix the error.

Obviously, this requires changing the error type, so you might want to look into how `libsyntax` handles it (there is also a `TODO` comment [here](https://github.com/rust-analyzer/rust-analyzer/blob/d77520fde3c953968beb09a3da80a0e7b17bbc04/crates/ra_syntax/src/parser_api.rs#L87) about this)

Reply:
Yeah, though this can be done in separate PR. Not sure if we have an infra for diagnostics with fixes in place yet.